### PR TITLE
[ImGui] Put config files into a folder

### PR DESCRIPTION
**Pull request by fredroy**
for less "pollution" in the working dir

### CODE REVIEW

**Review by alxbilger — COMMENTED** [review]
Should we wait for https://github.com/sofa-framework/sofa/pull/4875 to put the config files in the config folder?

**Comment by fredroy** [comment]
> Should we wait for [sofa-framework/sofa#4875](https://github.com/sofa-framework/sofa/pull/4875) to put the config files in the config folder?

Akshually 🤓 that was my first intention but I discovered the static function in BaseGui called setConfigDirectoryPath() which set globally the config dir so I wont use the function introduced in #4875 (in SofaGLFW at least)

**Comment by alxbilger** [comment]
So you want to use `BaseGUI::getConfigDirectoryPath()`?

File: SofaImGui/src/SofaImGui/AppIniFile.cpp [hunk 1]
```diff
@@ -21,16 +21,24 @@
...
 #include "AppIniFile.h"
 
+
 using namespace sofa;
 
 namespace sofaimgui
 {
+    const std::string& getConfigurationFolderPath()
+    {
+        static const std::string configPath = sofa::gui::common::BaseGUI::getConfigDirectoryPath() + "/imgui";
```
Comment (alxbilger) [inline SofaImGui/src/SofaImGui/AppIniFile.cpp:35]:
```suggestion
        static const std::string configPath = helper::system::FileSystem::append(sofa::gui::common::BaseGUI::getConfigDirectoryPath(), "imgui");
```
To ensure only one slash separator.

File: SofaImGui/src/SofaImGui/AppIniFile.cpp [hunk 2]
```diff
@@ -21,16 +21,24 @@
...
+        return configPath;
+    }
+
     const std::string& AppIniFile::getAppIniFile()
     {
-        static const std::string appIniFile(sofa::helper::Utils::getExecutableDirectory() + "/settings.ini");
+        static const std::string appIniFile(getConfigurationFolderPath() + "/settings.ini");
```
Comment (alxbilger) [inline SofaImGui/src/SofaImGui/AppIniFile.cpp:41]:
```suggestion
        static const std::string appIniFile(helper::system::FileSystem::append(getConfigurationFolderPath(), "settings.ini"));
```

Comment (alxbilger) [reply SofaImGui/src/SofaImGui/AppIniFile.cpp:41]: And also in ImGuiGUIEngine.cpp

File: SofaImGui/src/SofaImGui/windows/WindowState.cpp [hunk 3]
```diff
@@ -53,20 +54,32 @@ namespace windows {
...
+
         return stateStr == "OPEN";
     }
 
     void WindowState::writeState()
     {
-        std::ofstream file(m_filename);
+        const std::filesystem::path p = m_path;
+
+        if (!std::filesystem::exists(p.parent_path()))
```
Comment (alxbilger) [inline SofaImGui/src/SofaImGui/windows/WindowState.cpp:73]:
```suggestion
        if (!FileSystem::exists(p.parent_path()))
```

File: SofaImGui/src/SofaImGui/windows/WindowState.cpp [hunk 4]
```diff
@@ -53,20 +54,32 @@ namespace windows {
...
+        {
+            if (!std::filesystem::create_directories(p.parent_path()))
```
Comment (alxbilger) [inline SofaImGui/src/SofaImGui/windows/WindowState.cpp:75]:
```suggestion
            if (!FileSystem::createDirectory(p.parent_path()))
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

**Review by alxbilger — APPROVED** [review]

**Pull request merged**